### DISCUSSION TITLE
How to draw dendrogram only with annotation?

### DISCUSSION THREAD

**Discussion by ChuweiLin** [DingWB/PyComplexHeatmap#90] [Q&A]
I am using ver. 1.7.6.
I want to draw the dendrogram with annotation only, without heatmap. 
The DendrogramPlotter seems not have annotation.
The ClusterMapPlotter.plot_dendrograms needs row_order and col_order. I wonder how to use it.
Thank you so much!

**Comment by DingWB** [comment]
How can you calculate the dendrogram without the matrix (or data frame)?

I would suggest you plot the dendrogram, annotations, and heatmap together and then manually remove the main heatmap from the pdf.

**Reply by ChuweiLin** [reply]
I tried to remove the main heatmap from svg. Problem is the svg is too big to load. So I wonder if it is possible to just draw the dendrogram with annotation part, not the heatmap part.

**Reply by DingWB** [reply]
I see.
You can search on Google about how to clear an existing subplot, such as `cla` or `clf`.
For example, after running `cm=ClusterMapPlotter(**),` get the ax (cm.ax_heatmap) and clear the ax of the main heatmap.

You need to explore it yourself. I have never tried this.

**Reply by DingWB** [reply]
Another suggestion is to save it as a PDF (add parameter `rasterized=True`, rasterized will be set to True if the No. of rows or columns >=5000). Then, the main heatmap will be rasterized, and the file size will not be too large.

**Reply by ChuweiLin** [reply]
save as PDF and rasterized=True works. It's faster and smaller than svg. Thank you!
